### PR TITLE
Move notary demo driver-based code to test and minimise compile dependencies

### PR DESCRIPTION
**Pull request by andr3ej**
Also:
* Move demorun to test-utils
* Add module-module compile dependency graphs (requires graphviz)

### CODE REVIEW

File: constants.properties [hunk 1]
```diff
@@ -1,4 +1,4 @@
-gradlePluginsVersion=0.13.0
+gradlePluginsVersion=0.13.0+andr3ej-dependency-graphs
```
Comment (andr3ej) [inline constants.properties]: i will make this numeric before merging into master

File: graphs.gradle [hunk 2]
```diff
@@ -0,0 +1,75 @@
+class GraphProject {
```
Comment (adagys) [inline graphs.gradle]: could you provide a bit more context why we need all this?

Comment (andr3ej) [inline graphs.gradle]: IIRC the main motivation was that too many things have a compile dependency on test-utils (and thus all the things it rightly or wrongly has a compile dependency on) and I wanted to get a better idea of why. In the notary demo case, we can keep test-utils in test scope which forms the other half of this PR. And I wanted to know which modules are API from the POV of demos, for the guava removal work. Also highlighted is the compile dependency of demos on capsule/webcapsule which is not quite correct, and lots of redundant arcs that you won't see in gradle's built-in dependencies report.

Comment (adagys) [inline graphs.gradle]: sorry for the delay on this one – so the graph stuff is only a utility for visualising the dependencies right?

Comment (andr3ej) [inline graphs.gradle]: Yes that's right

**Comment by andr3ej** [comment]
I see that DummyContract has moved into test-utils, so we're now committed to all the samples having a compile dep on test-utils or part of it. So effectively all our demo code is in test scope, whereas here I was trying to have at least the flows etc in non-test. Maybe the right answer is a samples-common module. Anyway, as it's not going to work I'm going to close this PR and may open another with just the graphs stuff.

**Pull request closed without merging**